cluster/saltbase/salt/kube-addons/init.sls

```diff
@@ -101,13 +101,17 @@ addon-dir-create:
     - group: root
     - mode: 755
 
-{% if grains['os_family'] == 'RedHat' %}
+{% if pillar.get('is_systemd') %}
 
-/usr/lib/systemd/system/kube-addons.service:
+{{ pillar.get('systemd_system_path') }}/kube-addons.service:
```

Comment:
I will add; I don't think it is sufficient because of the daemon-reload salt issue, but I agree it is better to declare dependencies.